src/networkNode.ts

```diff
@@ -77,7 +83,26 @@ export class NetworkNode implements NetworkNodeInterface {
     args.push(cmd);
 
     const result = await client.runCommand(args, undefined, true);
-    return result.exitCode === 0;
+    if (result.exitCode === 0) {
+      // restart the port-fw if needed
+      const url = new URL(this.wsUri);
+      if (parseInt(url.port, 10) !== RPC_WS_PORT) {
+        const fwdPort = await client.startPortForwarding(
+          RPC_WS_PORT,
+          this.name,
+        );
+
+        this.wsUri = WS_URI_PATTERN.replace("{{IP}}", LOCALHOST).replace(
+          "{{PORT}}",
+          fwdPort.toString(),
+        );
+
+        this.apiInstance = undefined;
+      }
+
+      return true;
+    }
+    return false;
```

Comment:
Its better to do an early exit
```suggestion
    if(result.exitCode !== 0) return false;
    // restart the port-fw if needed
    const url = new URL(this.wsUri);
    if (parseInt(url.port, 10) !== RPC_WS_PORT) {
      const fwdPort = await client.startPortForwarding(
        RPC_WS_PORT,
        this.name,
      );

    this.wsUri = WS_URI_PATTERN.replace("{{IP}}", LOCALHOST).replace(
      "{{PORT}}",
      fwdPort.toString(),
    );

    this.apiInstance = undefined;
    return true;
```

Reply:
Yes, sounds better 👍